### PR TITLE
nixos/privacyidea: use envsubst to avoid leaking secrets to the store

### PR DESCRIPTION
**Pull request by Ma27**
###### Motivation for this change

Otherwise, you'd have to specify a few values I'd consider rather secret in the Nix store.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/security/privacyidea.nix [hunk 1]
```diff
@@ -206,14 +218,18 @@ in
...
           pi-manage = "${pkgs.sudo}/bin/sudo -u privacyidea -HE ${penv}/bin/pi-manage";
           pgsu = config.services.postgresql.superUser;
           psql = config.services.postgresql.package;
         in ''
           mkdir -p ${cfg.stateDir} /run/privacyidea
           chown ${cfg.user}:${cfg.group} -R ${cfg.stateDir} /run/privacyidea
+          ${lib.getBin pkgs.envsubst}/bin/envsubst -o ${cfg.stateDir}/privacyidea.cfg \
+                                                   -i "${piCfgFile}"
+          chown ${cfg.user}:${cfg.group} ${cfg.stateDir}/privacyidea.cfg
+          chmod o-r ${cfg.stateDir}/privacyidea.cfg
```
Comment (lheckemann) [inline nixos/modules/services/security/privacyidea.nix:232]: This is racy and can leak the secrets, prefer `umask 077` before creating the file or setting `o-x` on the parent directory.

File: nixos/tests/privacyidea.nix [hunk 2]
```diff
@@ -29,6 +35,12 @@ import ./make-test-python.nix ({ pkgs, ...} : rec {
     machine.start()
     machine.wait_for_unit("multi-user.target")
     machine.succeed("curl --fail http://localhost | grep privacyIDEA")
+    machine.succeed(
```
Comment (lheckemann) [inline nixos/tests/privacyidea.nix:38]: `grep "SECRET_KEY = 'testing'" /var/lib/privacyidea/privacyidea.cfg`?

File: nixos/modules/services/security/privacyidea.nix [hunk 3]
```diff
@@ -57,6 +57,18 @@ in
...
 
+      environmentFile = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "/root/privacyidea.env";
+        description = ''
+          File to load as environment file. Environment variables
+          from this file will be interpolated into the config file
+          using envsubst with this syntax:
+          <literal>$ENVIRONMENT ''${VARIABLE}</literal>
```
Comment (lheckemann) [inline nixos/modules/services/security/privacyidea.nix:68]: This example is confusing: it explains how to set the _other_ options in order to have them substituted in; I'd prefer a fuller explanation (with an example along the lines of `services.privacyidea = {secretKey = "$SECRET_KEY"; environmentFile = "/root/privacyidea.env";}` plus sample contents of `/root/privacyidea.env`), or just leaving it entirely up to the user to work it out (cut off after `… using envsubst.`)

**Review by lheckemann — CHANGES_REQUESTED** [review]
LGTM in principle, just a couple of changes :)

**Comment by Ma27** [comment]
@lheckemann thanks, fixed!

**Pull request merged**